src/cascade/input_data/configuration/builder.py

```diff
@@ -28,6 +60,214 @@ def initial_context_from_epiviz(configuration):
+    transform them, because each transformation is its own column for Dismod.
+    """
+    seen_covariate = defaultdict(set)
+    if configuration.country_covariate:
+        for covariate_configuration in configuration.country_covariate:
+            seen_covariate[covariate_configuration.country_covariate_id].add(covariate_configuration.transformation)
+
+    for cov_id, cov_transformations in seen_covariate.items():
+        yield cov_id, list(sorted(cov_transformations))
+
+
+def assign_covariates(model_context, configuration):
+    """
+    The EpiViz interface allows assigning a covariate with a transformation
+    to a specific target (rate, measure value, measure standard deviation).
+    There will be one Dismod-AT covariate for each (input covariate dataset,
+    transformation of that dataset).
+
+    Args:
+        model_context (ModelContext): model context that has age groups.
+            The context is modified by this function. Covariate columns are
+            added to input data and covariates are added to the list of
+            covariates.
+        configuration (Configuration): Holds settings from EpiViz form.
+
+    Returns:
+        function:
+            This function is a map from the covariate identifier in the
+            settings to the covariate name.
+
+    The important choices are assignment of covariates to observations and
+    integrands by
+    :py:func:`covariate_to_measurements_nearest_favoring_same_year`
+    and how reference values are chosen by
+    :py:func:`reference_value_for_covariate_mean_all_values`.
+    """
+    covariate_map = {}  # to find the covariates for covariate multipliers.
+    avgint_table = make_avgint_table(model_context, lambda x: "BradBellRocks")
+    age_groups = get_age_groups(model_context)
+
+    # This walks through all unique combinations of covariates and their
+    # transformations. Then, later, we apply them to particular target
+    # rates, meas_values, meas_stds.
+    for country_covariate_id, transforms in unique_country_covariate_transform(configuration):
+        demographics = dict(
+            age_group_ids="all",
+            year_ids="all",
+            sex_ids="all",
+            location_ids=[model_context.parameters.location_id],
+        )
+        ccov_df = country_covariates(country_covariate_id, demographics)
+        covariate_name = ccov_df.loc[0]["covariate_name_short"]
+
+        # There is an order dependency from whether we interpolate before we
+        # transform or transform before we interpolate.
+
+        # Decide how to take the given data and extend / subset / interpolate.
+        ccov_ranges_df = convert_gbd_ids_to_dismod_values(ccov_df, age_groups)
+
+        column_for_measurements = [
+            covariate_to_measurements_nearest_favoring_same_year(construct_column, ccov_ranges_df)
+            for construct_column in [model_context.input_data.observations, avgint_table]
```

Comment:
Ok. Here's where the alignment I was concerned about above comes in. I really want to make this more explicit. I'm not sure how though.